src/transformers/pipelines/automatic_speech_recognition.py

```diff
@@ -164,17 +209,27 @@ def _forward(self, model_inputs):
             tokens = self.model.generate(
                 encoder_outputs=encoder(**model_inputs), attention_mask=model_inputs.get("attention_mask")
             )
-            tokens = tokens.squeeze(0)
         elif model_class in MODEL_FOR_CTC_MAPPING.values():
+            stride = model_inputs.pop("stride", None)
             outputs = self.model(**model_inputs)
-            tokens = outputs.logits.squeeze(0).argmax(dim=-1)
+            tokens = outputs.logits.argmax(dim=-1)
+            if stride is not None:
+                left, right = stride
+                input_n = model_inputs["input_values"].shape[-1]
+                token_n = tokens.shape[-1]
+                left_token = int(left / input_n * token_n)
+                right_token = int((input_n - right) / input_n * token_n) + 1
+                tokens = tokens[:, left_token:right_token]
```

Comment:
this is the cleanest way of doing it I think @Narsil

Reply:
If we can avoid it I would rather not copy the tensors.